crates/matrix-sdk-common/src/linked_chunk/lazy_loader.rs

```diff
@@ -111,15 +112,25 @@ where
             });
         }
 
+        // Same check as before, but in reverse: the first chunk has a `lazy_previous`
+        // to the new first chunk.
+        if first_chunk.lazy_previous() != Some(new_first_chunk.identifier) {
+            return Err(LazyLoaderError::CannotConnectTwoChunks {
+                new_chunk: first_chunk.identifier,
+                with_chunk: new_first_chunk.identifier,
+            });
+        }
+
         // Alright. All checks are made.
     }
 
     // Insert the new first chunk.
     {
         // Transform the `RawChunk` into a `Chunk`.
         let lazy_previous = new_first_chunk.previous.take();
+        let new_first_chunk_identifier = new_first_chunk.identifier;
         let mut new_first_chunk =
-            Chunk::new_leaked(new_first_chunk.identifier, new_first_chunk.content);
+            Chunk::new_leaked(new_first_chunk_identifier, new_first_chunk.content);
```

Comment:
Could we revert these changes, and only keep what we had before? avoids one local variable, which is cognitive overhead

Reply:
oh, rebasing failure, thanks!